### PR TITLE
Delivery[PR] 6 - Sections 16, 17 and 18

### PR DESCRIPTION
**Pull request by NihilMaster**
In this Pull Request, I deliver sections 16, 17 and 18.

### CODE REVIEW

File: s18/use_Gadget.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require_relative "classes/gadget.rb"
```
Comment (mauriciosabogal) [inline s18/use_Gadget.rb:1]: What's the difference between `require` and `require_relative`?

File: s17/regexp.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+puts //.class
+
+s = 'The Ruby Programming Language is amazing'
+puts s =~ /y/
+puts /y/ =~ s
+p s =~ /x/   #=> nil
+puts s =~ /Ruby/ # position R
+
+
```
Comment (mauriciosabogal) [inline s17/regexp.rb:9]: Please don't leave all of these empty lines at the end of the file. Try to write cleaner code.

File: s18/use_Gadget.rb (hunk 1 above)
Comment (NihilMaster) [reply s18/use_Gadget.rb:1]: Both serve to call another file, but require has to cite the absolute path while requirer require_relative only the path from where the file is located from where it's called.

File: s17/regexp.rb (hunk 2 above)
Comment (NihilMaster) [reply s17/regexp.rb:9]: Sorry, this was because the Linux keyboard did not receive the sign ( ~ ) and I commented it at the end of the file.

**Review by mauriciosabogal — APPROVED** [review]

**Pull request merged**